Deepsource bugfixes (minor)

ElementUser opened the pull request:
# Deepsource bugfixes (minor)

Resolved the following Deepsource errors:
- PYL-R0205 (useless inheritance from `object` class)
- PTC-W0027 (`f-string` used without any expression)
- PYL-C0414 (import alias same as original package name)
- PTC-W0015 (unnecessary generator - rewrite as dict comprehension)
- PYL-R1720 (unnecessary `else/elif` used after raise)
- PYL-R1714 (performance-related: consider using `in`)

Commit:
Resolve Deepsource PYL-R1723 errors
- Reference: https://deepsource.io/gh/djuresic/rctbot-discord/issue/PYL-R1723/occurrences

Commit:
Resolved the following Deepsource errors:
- PYL-R0205 (useless inheritance from object class)
- PTC-W0027 (f-string used without any expression)
- PYL-C0414 (import aliasi same as original package name)
- PTC-W0015 (unnecessary generator - rewrite as dict comprehension)
- PYL-R1720 (unnecessary else/elif used after raise)

Commit:
Fixed "new" performance issue introduced (PYL-R1714)

djuresic commented on utils/phpserialize.py:
```diff
@@ -379,47 +379,47 @@ def _serialize(obj, keypos):
-            if obj is None:
-                return b"N;"
-            if isinstance(obj, bool):
-                return ("b:%i;" % obj).encode("latin1")
-            if isinstance(obj, (int, long)):
-                return ("i:%s;" % obj).encode("latin1")
-            if isinstance(obj, float):
-                return ("d:%s;" % obj).encode("latin1")
-            if isinstance(obj, basestring):
-                encoded_obj = obj
-                if isinstance(obj, unicode):
-                    encoded_obj = obj.encode(charset, errors)
-                s = BytesIO()
-                s.write(b"s:")
-                s.write(str(len(encoded_obj)).encode("latin1"))
-                s.write(b':"')
-                s.write(encoded_obj)
-                s.write(b'";')
-                return s.getvalue()
-            if isinstance(obj, (list, tuple, dict)):
-                out = []
-                if isinstance(obj, dict):
-                    iterable = obj.items()
-                else:
-                    iterable = enumerate(obj)
-                for key, value in iterable:
-                    out.append(_serialize(key, True))
-                    out.append(_serialize(value, False))
-                return b"".join(
-                    [b"a:", str(len(obj)).encode("latin1"), b":{", b"".join(out), b"}"]
-                )
-            if isinstance(obj, phpobject):
-                return (
-                    b"O"
-                    + _serialize(obj.__name__, True)[1:-1]
-                    + _serialize(obj.__php_vars__, False)[1:]
-                )
-            if object_hook is not None:
-                return _serialize(object_hook(obj), False)
-            raise TypeError("can't serialize %r" % type(obj))
```
This seems to have been purposefully made into an else statement so that it doesn't execute if keypos isn't falsy.

djuresic commented on utils/phpserialize.py:
```diff
@@ -379,47 +379,47 @@ def _serialize(obj, keypos):
                 return b's:0:"";'
             raise TypeError("can't serialize %r as key" % type(obj))
-        else:
-            if obj is None:
-                return b"N;"
```
This returns a different value in the if block; I'd keep the entire else block as is.

ElementUser replied:
Hmm, it looks like if `keypos` is truthy (line 365), then all the `if` branches under it has a `return` statement, so all conditional branches are accounted for.

If for some reason it reaches the line where it raises an error, then the function will be exited at that point too.

So I don't think the code reaches where the `else` block used to be if `keypos` is truthy or falsy.

djuresic replied:
You're right, all of them have a `return` statement, and a final `raise`.

The `else` block won't be reached so this is fine then.

djuresic reviewed (changes requested):
Changes to the else block in the phpserialize module should be reverted. Everything else looks good.

djuresic committed:
Reverted changes to the else block

djuresic committed:
Undo 1fafe93

- Reverted https://github.com/djuresic/rctbot-discord/pull/2/commits/1fafe937dcafe25615fb35c3f198f50dca95aa6b

Commit:
Merge pull request #1 from ElementUser/bugfix/deepsource_PYL-R1723

Merge bugfix/deepsource_PYL-R1723
